app/services/SessionAccessor.ts

```diff
@@ -32,4 +32,22 @@ export class SessionAccessor {
     this.sqlStorage.set('VP_TOKEN', token);
   }
 
+  getUsername(): Promise<string> {
+    if (this.localStorage.get('VP_USERNAME')) {
+      return PromiseWrapper.resolve(this.localStorage.get('VP_USERNAME'));
+    }
+
+    return this.sqlStorage.get('VP_USERNAME');
+  }
+
+  setUsername(username: string) {
+    if (!username) {
+      this.localStorage.remove('VP_USERNAME');
+      this.sqlStorage.remove('VP_USERNAME');
+      return;
+    }
+
+    this.localStorage.set('VP_USERNAME', username);
+    this.sqlStorage.set('VP_USERNAME', username);
```

Comment:
This is weird. The username is included inside of the token.

Take a look at `jwt-token` to see how you can decode a token.